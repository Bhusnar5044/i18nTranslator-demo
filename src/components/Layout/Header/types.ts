export type Props = {
    // Header prop type
    click: () => void;
};
