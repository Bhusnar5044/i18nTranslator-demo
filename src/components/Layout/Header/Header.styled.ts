import { styled } from 'styled-components';

export const HeaderContainer = styled('header')`
    display: flex;
    align-items: center;
    width: 100%;
    margin: auto;
    background-color: #008085;
`;

export const HeaderContent = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 90%;
    margin: auto;
`;

export const Logo = styled.img`
    width: 11.5rem;
    height: auto;
`;
