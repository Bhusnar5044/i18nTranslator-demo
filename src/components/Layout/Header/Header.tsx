import { SwitchLanguage } from '@components/SwitchLanguage/SwitchLanguage';
import { FC, memo } from 'react';
import { HeaderContainer, HeaderContent, Logo } from './Header.styled';

export const Header: FC = memo(() => {
    return (
        <HeaderContainer>
            <HeaderContent>
                <Logo src="/centime-logo.svg" />
                <SwitchLanguage />
            </HeaderContent>
        </HeaderContainer>
    );
});

Header.displayName = 'Header';
