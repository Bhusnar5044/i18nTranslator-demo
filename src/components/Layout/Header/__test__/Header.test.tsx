import { render, screen } from '@test-utils';
import { Header } from '../Header';

describe('Header', () => {
    it('should render properly', () => {
        const { container } = render(<Header />);
        expect(container).toMatchSnapshot();
        expect(screen.getByRole('img')).toBeInTheDocument();
    });
});
