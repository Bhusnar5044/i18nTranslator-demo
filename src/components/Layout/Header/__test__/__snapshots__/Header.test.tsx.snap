// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header > should render properly 1`] = `
<div>
  <header
    class="sc-hIFLYB cWfpsV"
  >
    <div
      class="sc-buuUZy cioRTw"
    >
      <img
        class="sc-ivAPlj hyHNBS"
        src="/centime-logo.svg"
      />
      <div
        class="form-group"
      >
        <select
          class="sc-eLqIir dmSHxF form-control"
        >
          <option
            class="sc-HzRMQ gxWMSa"
            disabled=""
            value=""
          >
            Switch Language
          </option>
        </select>
      </div>
    </div>
  </header>
</div>
`;

exports[`Header should render properly 1`] = `
<div>
  <header
    class="sc-iGsdBT fahxpC"
  >
    <div
      class="sc-ckeRpf bulpVB"
    >
      <img
        class="sc-kRYBTm gNfNKR"
        src="/centime-logo.svg"
      />
      <div
        class="sc-eLqIir dKTRNe"
        gap="1rem"
      >
        <select
          class="sc-dujCx iseyOr"
          title="single-select"
        >
          <option
            class="sc-bFmjqi gTeLlR"
            disabled=""
            value=""
          >
            Switch Language
          </option>
        </select>
      </div>
    </div>
  </header>
</div>
`;
